pinot-core/src/main/java/org/apache/pinot/core/segment/processing/framework/SegmentProcessorFramework.java

```diff
@@ -192,7 +194,10 @@ private List<File> doProcess()
         LOGGER.info("Start creating segments on partition: {}, numRows: {}, numSortFields: {}", partitionId, numRows,
             numSortFields);
         GenericRowFileRecordReader recordReader = fileReader.getRecordReader();
+        int maxNumRecordsPerSegment;
         for (int startRowId = 0; startRowId < numRows; startRowId += maxNumRecordsPerSegment, sequenceId++) {
+          maxNumRecordsPerSegment = (_segmentRowComputer != null) ? _segmentRowComputer.getNumRows()
```

Comment:
Instead of using `null`, we can have a default implementation that will return the value of `_segmentProcessorConfig.getSegmentConfig().getMaxNumRecordsPerSegment()`.

Reply:
done